webapi/src/main/scala/org/knora/webapi/util/ConstructResponseUtilV2.scala

```diff
@@ -345,17 +345,35 @@ object ConstructResponseUtilV2 {
             // the main resources point to dependent resources and would be treated as incoming links of dependent resources
             // this would create circular dependencies
 
-            val incomingLinks = incomingLinksForResource(resourceIri).filterNot {
+            // resources that point to this resource
+            val referringResources: Map[IRI, ResourceWithValueRdfData] = incomingLinksForResource(resourceIri).filterNot {
                 case (incomingResIri: IRI, assertions: ResourceWithValueRdfData) =>
                     alreadyTraversed(incomingResIri) || flatResourcesWithValues(incomingResIri).isMainResource
-            }.flatMap {
-                case (incomingResIri: IRI, assertions: ResourceWithValueRdfData) =>
-                    assertions.valuePropertyAssertions
             }
 
-            if (incomingLinks.nonEmpty) {
+            // link value assertions that point to this resource
+            val incomingLinkAssertions: Map[IRI, Seq[ValueRdfData]] = referringResources.values.foldLeft(Map.empty[IRI, Seq[ValueRdfData]]) {
```

Comment:
@benjamingeer This collects different instances of the same property rather than just overwriting them.